lib/lambda_ethereum_consensus/validator/duties.ex

```diff
@@ -157,15 +194,87 @@ defmodule LambdaEthereumConsensus.Validator.Duties do
     Map.put(duty, :subnet_id, subnet_id)
   end
 
+  ############################
+  # Accessors
+
+  @spec current_proposer(duties(), Types.epoch(), Types.slot()) :: proposer_duty() | nil
+  def current_proposer(duties, epoch, slot),
+    do: get_in(duties, [epoch, :proposers, slot])
+
+  @spec current_sync_committee(duties(), Types.epoch(), Types.slot()) ::
+          sync_committee_duties()
+  def current_sync_committee(duties, epoch, slot) do
+    for %{last_slot_broadcasted: last_slot} = duty <- sync_committee(duties, epoch),
+        last_slot < slot do
```

Comment:
Maybe we should add a comment explaining that this is to prevent broadcasting a message more than once in the same slot.

Reply:
Added!